### PR TITLE
Handle C/R in BasicImageReader (via reflection)

### PR DESCRIPTION
**Pull request by rvansa**
Fixes checkpoint failure due to open FD to modules on x86 (the Java code behaves differently on 32bit and 64bt).

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Reviewers
 * [Anton Kozlov](https://openjdk.org/census#akozlov) (@AntonKozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/crac.git pull/124/head:pull/124` \
`$ git checkout pull/124`

Update a local copy of the PR: \
`$ git checkout pull/124` \
`$ git pull https://git.openjdk.org/crac.git pull/124/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 124`

View PR using the GUI difftool: \
`$ git pr show -t 124`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/crac/pull/124.diff">https://git.openjdk.org/crac/pull/124.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/crac/pull/124#issuecomment-1750675959)

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/jimage/BasicImageReader.java [hunk 1]
```diff
@@ -103,30 +106,8 @@ protected BasicImageReader(Path path, ByteOrder byteOrder)
...
-                                 InvocationTargetException ex) {
-                            // fall thru - will only happen on JDK-8 systems where this code
-                            // is only used by tools using jrt-fs (non-critical.)
-                        }
-                    }
-
-                    return null;
-                }
-            });
+            channel = openFileChannel();
```
Comment (AntonKozlov) [inline src/java.base/share/classes/jdk/internal/jimage/BasicImageReader.java:109]: Could you avoid this refactoring to simplify future merges?

Comment (rvansa) [reply src/java.base/share/classes/jdk/internal/jimage/BasicImageReader.java:109]: Do you mean to just copy-paste the `openFileChannel()` code? Do we rather want any change in that code go unnoticed?

Comment (AntonKozlov) [reply src/java.base/share/classes/jdk/internal/jimage/BasicImageReader.java:109]: Oh! I see, you did refactoring to call openFileChannel() from registerIfCRaCPresent(). OK, that makes sense.

**Review by AntonKozlov — APPROVED** [review]

**Pull request closed without merging**